vendor/github.com/openshift/api/image/v1/types.go

```diff
@@ -417,6 +418,42 @@ type DockerImageReference struct {
 	ID string `protobuf:"bytes,5,opt,name=iD"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageStreamLayers describes information about the layers referenced by images in this
+// image stream.
+type ImageStreamLayers struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// blobs is a map of blob name to metadata about the blob.
+	Blobs map[string]ImageLayerData `json:"blobs" protobuf:"bytes,2,rep,name=blobs"`
```

Comment:
No maps in APIs.  I think I was finally convinced during the whole ordering thing we dealt with.  

If we must have map behavior internally, we have the ordered map implementation from cluster quota.